### PR TITLE
[options] Mergo doesn't play well with reference types

### LINKED ISSUE #2450: Panic when passing in custom logger.
### Description

Passing a custom logger with an exported struct member produces a panic on startup.

Running `wails dev` (or the built app):
```
2023/03/06 01:42:38 [INFO] Serving assets from frontend DevServer URL: http://localhost:5173/
panic: reflect: Field index out of range

goroutine 1 [running, locked to thread]:
reflect.Value.Field({0x10520e8e0, 0x105637d48, 0x199}, 0x0)
	/opt/homebrew/Cellar/go/1.20.1/libexec/src/reflect/value.go:1272 +0x194
github.com/imdario/mergo.deepMerge({0x10522f2a0, 0x1400000ed50, 0x199}, {0x10520e8e0, 0x105637d48, 0x199}, 0x14000149728, 0x3, 0x140000750c0)
	/Users/david/go/pkg/mod/github.com/imdario/mergo@v0.3.12/merge.go:93 +0x100c
github.com/imdario/mergo.deepMerge({0x105257c40, 0x1400000ed50, 0x16}, {0x105250080, 0x105637d48, 0x16}, 0x14000149728, 0x2, 0x140000750c0)
	/Users/david/go/pkg/mod/github.com/imdario/mergo@v0.3.12/merge.go:251 +0x6d8
github.com/imdario/mergo.deepMerge({0x105238040, 0x14000000588, 0x194}, {0x105238040, 0x1055f5a48, 0x194}, 0x14000149728, 0x1, 0x140000750c0)
	/Users/david/go/pkg/mod/github.com/imdario/mergo@v0.3.12/merge.go:272 +0x9a4
github.com/imdario/mergo.deepMerge({0x105286580, 0x14000000500, 0x199}, {0x105286580, 0x1055f59c0, 0x199}, 0x14000149728, 0x0, 0x140000750c0)
	/Users/david/go/pkg/mod/github.com/imdario/mergo@v0.3.12/merge.go:93 +0x1054
github.com/imdario/mergo.merge({0x1051d73c0, 0x14000000500}, {0x1051d73c0, 0x1055f59c0}, {0x0, 0x0, 0x0})
	/Users/david/go/pkg/mod/github.com/imdario/mergo@v0.3.12/merge.go:366 +0x344
github.com/imdario/mergo.Merge({0x1051d73c0, 0x14000000500}, {0x1051d73c0, 0x1055f59c0}, {0x0, 0x0, 0x0})
	/Users/david/go/pkg/mod/github.com/imdario/mergo@v0.3.12/merge.go:296 +0x58
github.com/wailsapp/wails/v2/pkg/options.MergeDefaults(0x14000000500)
	/Users/david/go/pkg/mod/github.com/wailsapp/wails/v2@v2.3.1/pkg/options/options.go:117 +0x4c
github.com/wailsapp/wails/v2/internal/app.CreateApp(0x14000000500)
	/Users/david/go/pkg/mod/github.com/wailsapp/wails/v2@v2.3.1/internal/app/app_dev.go:170 +0xc64
github.com/wailsapp/wails/v2/pkg/application.(*Application).Run(0x14000035940)
	/Users/david/go/pkg/mod/github.com/wailsapp/wails/v2@v2.3.1/pkg/application/application.go:65 +0x128
github.com/wailsapp/wails/v2.Run(0x14000000500)
	/Users/david/go/pkg/mod/github.com/wailsapp/wails/v2@v2.3.1/wails.go:14 +0x30
main.main()
	main.go:35 +0x3c8
```

### To Reproduce

I basically have a logs library that provides a custom logger that satisfies the wails interface. When I pass it in, I get a panic on startup. The only code in the stack trace that's mine is the `wails.Run` call from main.go. It took me a while, but I was able to narrow this down to my logger containing an exported struct member. **If I remove that member, the panic does not occur.**

```
package main

import (
	"embed"
	"runtime"

	"myApp/pkg/app"
	"myApp/pkg/logs"
	"github.com/wailsapp/wails/v2"
	"github.com/wailsapp/wails/v2/pkg/menu"
	"github.com/wailsapp/wails/v2/pkg/menu/keys"
	"github.com/wailsapp/wails/v2/pkg/options"
	"github.com/wailsapp/wails/v2/pkg/options/assetserver"
)

//go:embed all:frontend/dist
var assets embed.FS

func main() {
	// Create an instance of the app structure
	logger := logs.New()
	app := app.New(logger)
	appMenu := menu.NewMenu()
	FileMenu := appMenu.AddSubmenu("File")
	FileMenu.AddText("Quit", keys.CmdOrCtrl("q"), func(a *menu.CallbackData) {
		app.Quit()
	})

	if runtime.GOOS == "darwin" {
		// on macos, append EditMenu to enable Cmd+C,Cmd+V,Cmd+Z.
		appMenu.Append(menu.EditMenu())
	}

	// Create application with options
	err := wails.Run(&options.App{
		Title:       "myApp",
		Width:       1024,
		Height:      720,
		AssetServer: &assetserver.Options{Assets: assets},
		OnStartup:   app.Startup,
		Menu:        appMenu,
		Bind:        []interface{}{app},
		Logger:      logger,
	})
	if err != nil {
		panic("Error: " + err.Error())
	}
}
```

and the logs library (reduced to minimum possible):
```
package logs

import (
	"fmt"
	"log"
	"os"
)

// Logger satisfies logging interfaces.
type Logger struct {
	logger *log.Logger // Shares a Writer with InfoLog.
	Name   string      // throws a panic in wails v2.
}

// New returns a logger.
func New() *Logger {
	return &Logger{
		logger: log.New(os.Stdout, "", log.LstdFlags),
	}
}

func (l *Logger) Print(msg string) {
	l.writeMsg("[INFO] " + msg)
}
func (l *Logger) Trace(msg string) {
	l.writeMsg("[TRACE] " + msg)
}
func (l *Logger) Debug(msg string) {
	l.writeMsg("[DEBUG] " + msg)
}
func (l *Logger) Info(msg string) {
	l.writeMsg("[INFO] " + msg)
}
func (l *Logger) Warning(msg string) {
	l.writeMsg("[WARN] " + msg)
}
func (l *Logger) Error(msg string) {
	l.writeMsg("[ERROR] " + msg)
}
func (l *Logger) Fatal(msg string) {
	l.writeMsg("[FATAL] " + msg)
}

const callDepth = 2

func (l *Logger) writeMsg(msg string) {
	if err := l.logger.Output(callDepth, msg); err != nil {
		fmt.Println("Logger Error:", err) //nolint:forbidigo
	}
}
```

### Expected behaviour

No crash.

### Screenshots

_No response_

### Attempted Fixes

_No response_

### System Details

```shell
OS           | MacOS
Version      | 12.6
Go Version   | go1.19.5
Platform     | darwin
Architecture | arm64
Version      | v2.3.1
```
```


### Additional context

Thank you for any assistance!

### PR DESCRIPTION
**Pull request by stffabi**
Remove mergo and just use a simple hardcoded defaults approach.

Fixes #2450

### CODE REVIEW

File: v2/pkg/options/options.go [hunk 1]
```diff
@@ -148,7 +160,10 @@ func processMenus(appoptions *App) {
 	switch runtime.GOOS {
 	case "darwin":
 		if appoptions.Menu == nil {
-			appoptions.Menu = defaultMacMenu
+			appoptions.Menu = menu.NewMenuFromItems(
```
Comment (leaanthony) [inline v2/pkg/options/options.go:163]: Was this change intended? Just wondering how it affected the behaviour. Perhaps we should remove the default variables if they are causing an issue?

**Review by leaanthony — APPROVED** [review]
Thanks for jumping on this so quickly. Just a non blocking comment 👍👍

File: v2/pkg/options/options.go (hunk 1 above)
Comment (stffabi) [reply v2/pkg/options/options.go:163]: Yeah it was intended but not needed for the fix. It was just that after removing the default handling there was a `default.go` file with just the content of the `defaultMacMenu`. So I thought we could remove that and get rid of that almost empty `default.go` file.

**Comment by stffabi** [comment]
Thanks for approving 🙏

**Pull request merged**